RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerViewModel.cs

```diff
@@ -244,15 +244,15 @@ public ObservableCollection<CodeExplorerItemViewModel> Projects
             }
         }
 
-        private void ParserState_StateChanged(object sender, EventArgs e)
+        private void ParserState_StateChanged(object sender, ParserStateEventArgs e)
         {
             if (Projects == null)
             {
                 Projects = new ObservableCollection<CodeExplorerItemViewModel>();
             }
 
             IsBusy = _state.Status < ParserState.ResolvedDeclarations;
-            if (_state.Status != ParserState.ResolvedDeclarations)
+            if (e.State != ParserState.ResolvedDeclarations)
```

Comment:
Using `e.State` like this has caused issues when per-module parsing was asynchronous, a scenario that isn't completely ruled out but merely... postponed. Ideally the reading used in the comparison would be made thread-local off the *current* parser status:

    var state = _state.Status;

And then work off that. `ParserStateEventArgs` contains the parser state as it was read in *one* of the parser threads; at the instant in time that this handler runs, the *actual* parser state might compute differently; I think we could work to remove dependencies on that type, if there are any.